pkg/traceql/ast_execute.go

```diff
@@ -99,6 +98,72 @@ func (o SpansetOperation) evaluate(input []*Spanset) (output []*Spanset, err err
 				output = append(output, matchingSpanset)
 			}
 
+		case OpSpansetDescendant:
+			if len(lhs) > 0 && len(rhs) > 0 {
+				// Clone here to capture previously computed aggregates, grouped attrs, etc.
+				matchingSpanset := input[i].clone()
+				matchingSpanset.Spans = nil
+				// TODO: In what situations do lhs and rhs have more than one spanset,
+				// and what should the result be?
+				for _, l := range lhs[0].Spans {
```

Comment:
should we fail in the event of multiple spansets? i can see the same for the other operators.

"spanset operators are not supported for multiple spansets per trace. consider using coalesce()"

?

Reply:
Agree.  When we see a query and use case that requires this, then hopefully we can also make a determination on what the output should be algorithmically.